src/output/output_hls.cpp

```diff
@@ -233,6 +233,9 @@ namespace Mist{
   void OutHLS::onHTTP(){
     initialize();
     bootMsOffset = 0;
+    HTTP::Parser request = H;
+    H.Clean();
```

Comment:
In `golang` we could use `defer H.Clean()`

In c++17 we can create `defer` to look like this:
```
    DeferCode cleanHeaders([&]() { H.Clean(); });
```

We define defer as:
```
template <typename Finalizer>
class DeferCode {
private:
  Finalizer finalizer;

public:
  DeferCode(Finalizer&& lambda)
      : finalizer(std::forward<Finalizer>(lambda)) { }

  ~DeferCode() {
    try {
      finalizer();
    } catch(...) {
      // no exceptions allowed to escape
    }
  }
};
```